### PR TITLE
feat: POST judgingSessions

### PR DESCRIPTION
**Pull request by granthill7**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### CODE REVIEW

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

File: packages/api/tests/api/expoJudgingSession/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+describe('/expoJudgingSession post endpoint registration', () => {
+  it('uses adminMiddleware and registers the route for the handler', async () => {
+    await jest.isolateModulesAsync(async () => {
+      const { expoJudgingSession } = await import('../../../src/api/expoJudgingSession');
+
+      const app = express();
+      app.use(expoJudgingSession);
+      const res = await supertest(app).post('');
+      expect(res.status).toEqual(200);
```
[the remark this answers is not in the document]
Comment (granthill7) [reply packages/api/tests/api/expoJudgingSession/index.test.ts:22]: Is the post endpoint not being called by `supertest(app).post('');`?

File: packages/shared/src/schema/expoJudgingSession/post.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { post } from '../judgingSession';
+
+export const expoJudgingSessionPost = post.extend({});
```
[the remark this answers is not in the document]
Comment (granthill7) [reply packages/shared/src/schema/expoJudgingSession/post.ts:3]: Is that not what it's doing? `import { post } from '../judgingSession';`. I thought any additional things we want to add later will go into that {} object

File: packages/shared/src/schema/judgingSession/post.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { z } from 'zod';
+
+export const post = z.object({
+  inviteCode: z.string().uuid(),
```
[the remark this answers is not in the document]
Comment (granthill7) [reply packages/shared/src/schema/judgingSession/post.ts:4]: How can I validate that? I think v4 UUIDs are all structured the same way

File: packages/api/tests/api/expoJudgingSession/index.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (SpencerKaiser) [reply packages/api/tests/api/expoJudgingSession/index.test.ts:30]: It is being called by that line but you aren't verifying that it is... if you comment out the line that registers the post endpoint, this test may still succeed. It's just being explicit and guaranteeing the route handler we expect to be called is actually called

File: packages/shared/src/schema/expoJudgingSession/post.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (SpencerKaiser) [reply packages/shared/src/schema/expoJudgingSession/post.ts:3]: I'm an idiot; I misread this 😓

File: packages/shared/src/schema/judgingSession/post.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (SpencerKaiser) [reply packages/shared/src/schema/judgingSession/post.ts:4]: [You did 😅 ](https://github.com/AmericanAirlines/Hangar/pull/506/files#diff-72016482ad3646b06518fa2bee60469e9b4c441f5ee6e1a8b0b899dce66c2974R17)

File: packages/api/src/api/expoJudgingSession/post.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  let expoJudgingSession: ExpoJudgingSession | undefined;
+
+  try {
+    expoJudgingSession = new ExpoJudgingSession({
+      createdBy: user.toReference(),
+    });
+    await entityManager.persistAndFlush(expoJudgingSession);
+  } catch (error) {
+    // All other errors
```
Comment (SpencerKaiser) [inline packages/api/src/api/expoJudgingSession/post.ts:25]:
```suggestion
```

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**